BitFaster.Caching/Synchronized/Atomic.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BitFaster.Caching.Synchronized
+{
+    [DebuggerDisplay("IsValueCreated={IsValueCreated}, Value={ValueIfCreated}")]
+    public class Atomic<K, V>
```

Comment:
This is probably the wrong name - it's like indivisible smallest unit rather than exactly once.

Probably it should be called:

- Synchronized
- ScopedSynchronized
- AsyncSynchronized
- ScopedAsyncSynchronized

It's kind of weird having AsyncSynchronized etc. though.